crates/ruff/src/args.rs

```diff
@@ -437,6 +439,28 @@ pub struct FormatCommand {
     preview: bool,
     #[clap(long, overrides_with("preview"), hide = true)]
     no_preview: bool,
+
+    /// Format code starting at the given character offset (zero based).
+    ///
+    /// When specified, Ruff will try to only format the code after the specified offset but
+    /// it might be necessary to extend the start backwards, e.g. to the start of the logical line.
+    ///
+    /// Defaults to the start of the document.
+    ///
+    /// The option can only be used when formatting a single file. Range formatting of notebooks is unsupported.
+    #[arg(long)]
+    pub range_start: Option<usize>,
```

Comment:
What if we used `range_start_char` and `range_end_char`? Would that help future-proof the API?

Reply:
What about `range_start_codepoint` instead? (Since I think "character" is ambiguous here.) It's a little long, but it's probably better to start longer and come up with shorter names later if new use cases arise.

Reply:
Resolved, because it's now a single option